### ISSUE TITLE
Google: could not find default credentials

### ISSUE THREAD

**Issue by prestonvanloon** [circleci/docker-hello-google#5]
deploy.sh fails with 

error: google: could not find default credentials. See https://developers.google.com/accounts/docs/application-default-credentials for more information.

Not sure what to do. Followed the tutorial exactly

**Comment by zzak** [comment]
Hello @prestonvanloon,

Did you follow [this guide](https://circleci.com/docs/continuous-deployment-with-google-container-engine/)?

**Comment by prestonvanloon** [comment]
Yep. That's the one. I had to use ACCT_AUTH as the environment variable first, but now I'm stuck at this error I reported.

**Comment by felicianotech** [comment]
Which did you follow more? The actual project or the guide? They have diverged slightly in terms of their suggestions. For example, the readme says to set the envar `ACCT_AUTH` however if you follow the guide, the new suggestion is to set `GCLOUD_SERVICE_KEY` instead.

At the end of the day, it's just a name and as long as you use it correctly, that matters most.

**Comment by prestonvanloon** [comment]
That isn't the issue. I was just bringing up that there are descripencies in the repository that is supposed to be used to follow along the guide. 

I followed the guide using the correct environment variable name and got this error. This error is different than one caused by not setting the environment variable properly.

**Comment by justinwp** [comment]
same issue. was already using gcloud for other tasks such as pushing/pulling from gcr.io

**Comment by andersgb** [comment]
Had the same issue, but it seems to work now following a suggestion from https://github.com/google/google-auth-library-ruby/issues/65 

Apparently the environment variable GOOGLE_APPLICATION_CREDENTIALS should point to the json file with the service account login. 
I added the following line under `machine: environment:` in circle.yml:
`GOOGLE_APPLICATION_CREDENTIALS: ${HOME}/account-auth.json`

**Comment by prestonvanloon** [comment]
@andersgb That should probably go in the circle.yml, IMO. It does solve the problem for me too.

**Comment by juanpastas** [comment]
I had the same problem. I solved it with:

```
gcloud config set container/use_client_certificate True
```

From: https://github.com/kubernetes/kubernetes/issues/30617.

**Comment by arekkas** [comment]
Please add this to the guide

**Comment by naderg-clewmed** [comment]
I had same issue and solved it with:

`gcloud auth application-default login`

**Comment by SweetDreamEng** [comment]
How can I find gloud?

**Comment by Pratik521992** [comment]
> How can I find gloud?

Its should be available when you install Google cloud SDK
